IPython/terminal/pt_inputhooks/qt.py

```diff
@@ -1,15 +1,31 @@
 import sys
+import os
 from IPython.external.qt_for_kernel import QtCore, QtGui
 
 # If we create a QApplication, keep a reference to it so that it doesn't get
 # garbage collected.
 _appref = None
-
+_already_warned = False
 
 def inputhook(context):
     global _appref
     app = QtCore.QCoreApplication.instance()
     if not app:
+        if sys.platform == 'linux':
+            try:
+                # DISPLAY or WAYLAND_DISPLAY is set and not empty
+                assert os.environ.get('DISPLAY') or os.environ.get('WAYLAND_DISPLAY')
```

Comment:
Why not use `if...else` instead of `assert` here?

Reply:
I find the logic easier. (There should be no runtime difference.)
```python
if not (os.environ.get('DISPLAY') or os.environ.get('WAYLAND_DISPLAY')):
    ...
# vs 
try:
    assert os.environ.get('DISPLAY') or os.environ.get('WAYLAND_DISPLAY'))
except Exception:
    ...
````
But if you prefare `if`, I can change the code.

Reply:
I'd prefer an `if`; it feels strange to throw an error only to immediately catch it again.

Reply:
ok, I changed it to if